### ISSUE TITLE
[Bug?]: React Utils tree shaking issue with commonJS

### ISSUE THREAD

**Issue by caseybaggz** [pluralsight/pando#1093]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When using the `next` version of React utils in a Server-side setup (NextJS), there is the following error that seems to be around the topic of dependency management/expectations via the rollup tree-shaking result (also might be a bug in the tree-shaking function in our config):

```bash
Unhandled Runtime Error

TypeError: _vendored_react_jsx_runtime_js_commonjs_module_339116cb_js__WEBPACK_IMPORTED_MODULE_3__.j.exports.jsx is not a function
Call Stack
TabsProvider
node_modules/@pluralsight/react-utils/npm/browser/src/hooks/useTabs/context.tsx-c3b1ce26.js (18:22)
```

Basically, Next is confused about where to source the React JSX function which means it's related to meta created from our Rollup process.

**ES build result works as expected. This is only a commonJS build problem.**

### Expected behavior 🤔

CommonJS build should produce a result that does not yield any errors in the local install of the project.

### Steps to reproduce 🕹

Steps:

1. Install NextJS project
2. Install RU package
3. Try and use the Tabs hooks
4. GET THE ERROR (the only API that produces a React component)

### Context 🔦

React-utils, rollup-up, commonJS, tree-shaking

**Comment by caseybaggz** [comment]
Confirmed the nullish plugin is being used (another babel-related issue) via `debug` option:
```bash
~/g/pando ❯❯❯ yarn workspace @pluralsight/react-utils run build

index.experimental.js → npm/browser, npm/node...
@babel/preset-env: `DEBUG` option

Using targets:
{
  "node": "18.4"
}

Using modules transform: auto

Using plugins:
  syntax-class-static-block
  syntax-private-property-in-object
  syntax-class-properties
  syntax-numeric-separator
  syntax-nullish-coalescing-operator 👈
  syntax-optional-chaining
  syntax-json-strings
  syntax-optional-catch-binding
  syntax-async-generators
  syntax-object-rest-spread
  syntax-dynamic-import
  syntax-export-namespace-from
  syntax-top-level-await

Using polyfills: No polyfills were added, since the `useBuiltIns` option was not set.
```

**Comment by caseybaggz** [comment]
Still getting error in latest next release using NextJS - reveals a little more info:

```bash
Unhandled Runtime Error
TypeError: _vendored_react_jsx_runtime_js_commonjs_module_339116cb_js__WEBPACK_IMPORTED_MODULE_3__.j.exports.jsx is not a function

Call Stack
TabsProvider
node_modules/@pluralsight/react-utils/npm/browser/src/hooks/useTabs/context.tsx-c3b1ce26.js (18:22)
renderWithHooks
node_modules/react-dom/cjs/react-dom.development.js (16305:0)
mountIndeterminateComponent
node_modules/react-dom/cjs/react-dom.development.js (20074:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (21587:0)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (4164:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (4213:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4277:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (27451:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (26557:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (26466:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (26434:0)
performSyncWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (26085:0)
flushSyncCallbacks
node_modules/react-dom/cjs/react-dom.development.js (12042:0)
flushSync
node_modules/react-dom/cjs/react-dom.development.js (26201:0)
Object.scheduleRefresh
node_modules/react-dom/cjs/react-dom.development.js (27795:0)
eval
node_modules/next/dist/compiled/react-refresh/cjs/react-refresh-runtime.development.js (265:0)
Set.forEach
<anonymous>
Object.performReactRefresh
node_modules/next/dist/compiled/react-refresh/cjs/react-refresh-runtime.development.js (254:0)
applyUpdate
node_modules/next/dist/compiled/@next/react-refresh-utils/dist/internal/helpers.js (123:0)
statusHandler
node_modules/next/dist/compiled/@next/react-refresh-utils/dist/internal/helpers.js (140:0)
setStatus
file:///Users/casey-baggz/Desktop/next-test/.next/static/chunks/webpack.js (453:55)
<unknown>
/_next/static/chunks/webpack.js (624:21)
```
